libr/core/cmd_flag.c

```diff
@@ -341,6 +341,107 @@ static bool adjust_offset(RFlagItem *flag, void *user) {
 	return true;
 }
 
+static int flag_space_list(RFlag *f, int mode) {
+	r_return_val_if_fail (f, -1);
+	const char *defspace = NULL;
+	int count, len, i, j = 0;
+	bool allSelected = f->space_idx == -1;
+	if (mode == 'j') {
+		r_cons_printf ("[");
+	}
+	for (i = 0; i < R_FLAG_SPACES_MAX; i++) {
+		if (!f->spaces[i]) {
+			continue;
+		}
+		count = r_flag_space_count (f, i);
+		if (mode == 'q') {
```

Comment:
please let's do these changes in a separate PR. if you notice, i just moved the code from flag/flag.c, because util/RSpaces doesn't have cb_printf and there's no need for it to have cb_printf.

Reply:
I dont get whats the relationship between chaning 4 ifelses for 1 switch to printf

Reply:
the relationship is that i moved this code from one file to another. If i have to fix all shitty things in all pieces of code I move every time, I would start trying to fix X and I would end up fixing Y, Z but not X.